### PR TITLE
HDDS-1926. The new caching layer is used for old OM requests but not updated.

### PR DESCRIPTION
**Pull request by bharatviswa504**
When I am trying to add a test with restart manager, and then try to create the same volume, I am getting some NetUtils EOF exception. Will try to see why it is causing, if I am able to solve this issue, will post an Integration test for the scenario mentioned in the Jira.

### CODE REVIEW

**Comment by bharatviswa504** [comment]
Added tests with OM restart.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 79 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 627 | trunk passed |
| +1 | compile | 368 | trunk passed |
| +1 | checkstyle | 72 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 968 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 172 | trunk passed |
| 0 | spotbugs | 479 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 695 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 564 | the patch passed |
| +1 | compile | 385 | the patch passed |
| +1 | javac | 385 | the patch passed |
| +1 | checkstyle | 84 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 765 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 171 | the patch passed |
| +1 | findbugs | 718 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 358 | hadoop-hdds in the patch passed. |
| -1 | unit | 2156 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 41 | The patch does not generate ASF License warnings. |
| | | 8391 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineCreateAndDestory |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1247/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1247 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 4ae2dd5a127b 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 827dbb1 |
| Default Java | 1.8.0_222 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1247/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1247/1/testReport/ |
| Max. process+thread count | 4666 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/ozone-manager U: hadoop-ozone/ozone-manager |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1247/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 74 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 1 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 13 | Maven dependency ordering for branch |
| +1 | mvninstall | 621 | trunk passed |
| +1 | compile | 404 | trunk passed |
| +1 | checkstyle | 80 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 976 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 171 | trunk passed |
| 0 | spotbugs | 448 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 669 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 21 | Maven dependency ordering for patch |
| +1 | mvninstall | 593 | the patch passed |
| +1 | compile | 415 | the patch passed |
| +1 | javac | 415 | the patch passed |
| +1 | checkstyle | 83 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 802 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 188 | the patch passed |
| +1 | findbugs | 742 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 396 | hadoop-hdds in the patch failed. |
| -1 | unit | 2268 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 43 | The patch does not generate ASF License warnings. |
| | | 8727 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.container.TestReplicationManager |
|   | hadoop.ozone.client.rpc.TestCommitWatcher |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineProvider |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineCreateAndDestory |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |
|   | hadoop.ozone.client.rpc.TestWatchForCommit |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1247/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1247 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 660aef69b18d 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 827dbb1 |
| Default Java | 1.8.0_222 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1247/2/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1247/2/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1247/2/testReport/ |
| Max. process+thread count | 4639 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/ozone-manager hadoop-ozone/integration-test U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1247/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java [hunk 1]
```diff
@@ -269,14 +276,23 @@ protected void initializeOmTables() throws IOException {
     userTable =
         this.store.getTable(USER_TABLE, String.class, VolumeList.class);
     checkTableStatus(userTable, USER_TABLE);
+
+    // As now we have eviction policies, and for non-HA code path we don't
+    // support cache and cleanup policies setting cache to manual.
+    TableCacheImpl.CacheCleanupPolicy cleanupPolicy;
+    if (isRatisEnabled) {
+      cleanupPolicy = TableCacheImpl.CacheCleanupPolicy.NEVER;
```
Comment (arp7) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java]: Let's use ternary operator here.

Comment (bharatviswa504) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java]: Done.

**Comment by bharatviswa504** [comment]
Thank You @arp7 for the review.
Addressed review comments.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 36 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 35 | Maven dependency ordering for branch |
| +1 | mvninstall | 627 | trunk passed |
| +1 | compile | 406 | trunk passed |
| +1 | checkstyle | 79 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 958 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 163 | trunk passed |
| 0 | spotbugs | 428 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 624 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 20 | Maven dependency ordering for patch |
| +1 | mvninstall | 546 | the patch passed |
| +1 | compile | 367 | the patch passed |
| +1 | javac | 367 | the patch passed |
| +1 | checkstyle | 75 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 743 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 164 | the patch passed |
| +1 | findbugs | 706 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 345 | hadoop-hdds in the patch passed. |
| -1 | unit | 1523 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 41 | The patch does not generate ASF License warnings. |
| | | 7614 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1247/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1247 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux f7b2e3bcf499 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 3cc0ace |
| Default Java | 1.8.0_222 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1247/3/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1247/3/testReport/ |
| Max. process+thread count | 4256 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/ozone-manager hadoop-ozone/integration-test U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1247/3/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 33 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 12 | Maven dependency ordering for branch |
| +1 | mvninstall | 594 | trunk passed |
| +1 | compile | 385 | trunk passed |
| +1 | checkstyle | 71 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 857 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 171 | trunk passed |
| 0 | spotbugs | 467 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 683 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 19 | Maven dependency ordering for patch |
| +1 | mvninstall | 563 | the patch passed |
| +1 | compile | 387 | the patch passed |
| +1 | javac | 387 | the patch passed |
| +1 | checkstyle | 76 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 727 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 178 | the patch passed |
| +1 | findbugs | 746 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 347 | hadoop-hdds in the patch passed. |
| -1 | unit | 1712 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 43 | The patch does not generate ASF License warnings. |
| | | 7755 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.om.TestKeyManagerImpl |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1247/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1247 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 205fbcee810f 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 00b5a27 |
| Default Java | 1.8.0_222 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1247/4/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1247/4/testReport/ |
| Max. process+thread count | 4965 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/ozone-manager hadoop-ozone/integration-test U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1247/4/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request closed without merging**